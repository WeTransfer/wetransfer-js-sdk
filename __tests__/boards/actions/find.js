const routes = require('../../../src/config/routes');
const findAction = require('../../../src/boards/actions/find');
const WTError = require('../../../src/error');

describe('Find board action', () => {
  let find = null;
  const mocks = {};
  beforeEach(() => {
    mocks.request = { send: jest.fn() };
    mocks.request.send.mockReturnValue({
      id: 'random-hash',
      name: 'Little kittens',
      description: null,
      state: 'uploading',
      url: 'https://we.tl/s-random-hash',
      items: [],
    });

    find = findAction({
      routes,
      request: mocks.request,
    });
  });

  it('should create a find board request', async () => {
    const board = await find('random-hash');
    expect(board).toMatchSnapshot();
  });

  it('should throw an exception if the request fails', async () => {
    mocks.request.send.mockImplementation(() =>
      Promise.reject(new Error('Network error.'))
    );

    try {
      await find('random-hash');
    } catch (error) {
      expect(error).toBeInstanceOf(WTError);
    }
  });
});
