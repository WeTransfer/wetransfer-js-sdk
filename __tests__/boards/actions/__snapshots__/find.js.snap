// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find board action should create a find board request 1`] = `
RemoteBoard {
  "description": null,
  "id": "random-hash",
  "items": Array [],
  "name": "Little kittens",
  "state": "uploading",
  "url": "https://we.tl/s-random-hash",
}
`;
